### DISCUSSION TITLE
Unmount component when pathname is changed

### DISCUSSION THREAD

**Discussion by ariran5** [vercel/next.js#23266] [Help]
### What version of Next.js are you using?

10.0.9

### What version of Node.js are you using?

14.15.3

### What browser are you using?

Safari

### What operating system are you using?

macOS

### How are you deploying your application?

nom run dev

### Describe the Bug

If I use component on 2 pages(page '/' and page '/qwe') and I click to Link from '/' to page '/qwe', component will be unmounted. When page /qwe will be loaded, component call mounted callback again. 

its bad for me, because I use animation for show site menu. After change page menu animation shows again.

### Expected Behavior

if the component is on the page then it should not be redrawn

### To Reproduce

For example:
```js
// Component.js
import { useEffect } from 'react'
export default () => {
  useEffect(() => {
    return () => {
       // called on each change page, its bad, because if I want call useEffect on new page
       // I can use array of dependencies useEffect(..., [pathname])
      console.log(1)

    }
  }, [])
  return <div>Content</div>
}
```

```js
// index page 
import Link from 'next/link'
import Component1 from '../Component'

export default function Home() {
  return (
    <div >
      <Component1 />
      <Link href="/qwe">
        <a>to /qwe</a>
      </Link>
    </div>
  )
}

```

```js
// qwe page

import Link from 'next/link'
import Component1 from '../Component'

export default function Home() {
  return (
    <div >
      <Component1 />
      <Link href="/">
        <a>to /</a>
      </Link>
    </div>
  )
}

```

**Comment by balazsorban44** [comment]
Read this:

https://adamwathan.me/2019/10/17/persistent-layout-patterns-in-nextjs/

**Reply by ariran5** [reply]
why is the current solution considered good? why do I need to use this method to fix this feature ?

**Reply by ariran5** [reply]
i think, its bad feature, because, if i whant fetch data for site menu in useEffect callback, i need to use strange solutions for fix it

**Reply by ariran5** [reply]
Sory. its react problem
